contract/tests/test_contract.py

```diff
@@ -303,8 +303,8 @@ def test_contract_daily(self):
         self.assertEqual(self.acct_line.last_date_invoiced, last_date_invoiced)
 
     def test_contract_invoice_followers(self):
-        self.acct_line.recurring_next_date = "2018-02-23"
         self.acct_line.recurring_rule_type = "daily"
+        self.acct_line.recurring_next_date = "2018-02-23"
```

Comment:
Why changing this order?

Reply:
The fix is to recompute  recurring_next_date when we change recurring_rule_type. So to force recurring_next_date we must set it after recurring_rule_type otherwise the date is not keeped.

Reply:
Then this seems a problem in that code

Reply:
But this shouldn't have nothing to do with what you put on the title of the PR. Keep the diff minimum.

Reply:
The order is important otherwise tests are not OK

Reply:
But they were OK before your PR, so then what is modifying seems not OK, or at least not with an explanation of the steps to reproduce the problem. And the ideal situation is to not depend in any input order.

Reply:
I don't know how to explain. Because my PR fix the onchange method. So it's logic that the value computed by this method is impacted !

Reply:
@pedrobaeza could you validate this pr ?